### PR TITLE
Feature create `Button` component

### PR DESCRIPTION
**Pull request by daoanhhuy26012001**
- [x] Create the `Button` component
- [x] Split object into a storybook
- [x] Set style component
- **NOTE**:
  - The three component objects I create in the storybook will correspond to: image, image, image
  - See more details: [Link](https://docs.google.com/document/d/1iTrT9cdKrMCjUc6aDLS514gEk7BUFj-m2SCs2wCy2Lw/edit#heading=h.7t5pbcmau6ih)
- RESULT:
image

### CODE REVIEW

**Comment by phattruong-agilityio** [comment]
Could you please double check the code style. we are required `;` at the end of statement and format code before push

**Comment by phattruong-agilityio** [comment]
Btw, Please double check the commit message, please more detail and make it more clearer.

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * button component
+ *
+ * @param {boolean} primary - Is this the principal call to action on the page?
+ * @param {string} backgroundColor - What background color to use
+ * @param {string} size - How large should the button be? (Options: 'small', 'medium', 'large')
+ * @param {string} label - button contents (required)
+ * @param {function} onClick - Optional click handler
+ */
+
+export const Button = ({ primary, backgroundColor, size, label, ...props }) => {
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:14]: Default value for props

Comment (phattruong-agilityio) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx:14]: Boolean should be `isSomething`, `hasSomething`, .... I remember told you before, Please keep the naming practice

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const mode = primary ? 'button--primary' : 'button--secondary'
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:15]: It's not a `mode`. It's variant

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <button
+      type="button"
+      className={['button', `button--${size}`, mode].join(' ')}
+      style={backgroundColor && { backgroundColor }}
+      {...props}>
+      {label}
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:22]: using `children` is better

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Default props for button component
+ *
+ * @prop {string|null} backgroundColor - Default background color (null if not specified)
+ * @prop {boolean} primary - Default value for the primary prop (false if not specified)
+ * @prop {string} size - Default size for the button (Options: 'small', 'medium', 'large')
+ * @prop {undefined} onClick - Default value for the onClick prop (undefined if not specified)
+ */
+Button.defaultProps = {
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:35]: Do not do defaultProps here

File: ReactJS/practice-one/.storybook/preview.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import '../src/styles/main.css'
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/.storybook/preview.js:1]: Please add prefix import
```js
// Styles
import '../src/styles/main.css'

```

File: ReactJS/practice-one/src/components/Common/Button/Button.stories.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { Button } from '.'
```
[the remark this answers is not in the document]
Comment (phattruong-agilityio) [reply ReactJS/practice-one/src/components/Common/Button/Button.stories.jsx:1]: Also update end of line with `;`

[the remark this answers is not in the document]
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/Button.stories.jsx:1]: Done

File: ReactJS/practice-one/src/components/Common/Button/index.jsx (hunk 4 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx:35]: Done

File: ReactJS/practice-one/src/components/Common/Button/index.jsx (hunk 2 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx:15]: Done

File: ReactJS/practice-one/src/components/Common/Button/index.jsx (hunk 3 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx]: Done

File: ReactJS/practice-one/src/components/Common/Button/index.jsx (hunk 1 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx]: Done

File: ReactJS/practice-one/.storybook/preview.js (hunk 5 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/.storybook/preview.js]: Done

File: ReactJS/practice-one/src/components/Common/Button/Button.stories.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/Button.stories.jsx]: done

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+// Import the CSS file
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:1]: Style

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+ * button component
+ *
+ * @param {boolean} primary - Is this the principal call to action on the page?
+ * @param {string} backgroundColor - What background color to use
+ * @param {string} size - How large should the button be? (Options: 'small', 'medium', 'large')
+ * @param {string} children - button contents (required)
+ * @param {function} onClick - Optional click handler
+ */
+
+export const Button = ({
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/Button/index.jsx:14]: Missing default props

Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx:14]: Done <3

File: ReactJS/practice-one/src/components/Common/Button/index.jsx (hunk 7 above)
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx]: Done

File: ReactJS/practice-one/src/constants/constant.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+// SIZE
```
Comment (phuocduong-agilityio) [inline ReactJS/practice-one/src/constants/constant.js:1]: Consider to build constant by object. So we can use it for ex: BTN.MEDIUM_SIZE or something else like that

File: ReactJS/practice-one/src/components/Common/Button/index.jsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * @param {function} onClick - Optional click handler
+ */
+
+export const Button = ({ variant, size = SIZE_MEDIUM, children, onClick }) => {
+  return (
+    <button
+      type="button"
+      onClick={onClick}
+      className={['button', `button--${size}`, `button--${variant}`].join(' ')}>
+      {children}
```
[the remark this answers is not in the document]
Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/Button/index.jsx:20]: Done

**Pull request merged**